# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1375396489 14400
#      Thu Aug 01 18:34:49 2013 -0400
# Node ID 5014205810876a5315b76234eb86357d7d302b7c
# Parent  743c9745009afc373a88931dc90e38998d8ad345
Fix the figure propety "currentpoint" for the FLTK toolkit.

* libinterp/dldfcn/__init_fltk__.cc: Fix error is calcualting the current
point for the FLTK toolkit (set_currentpoint ()).

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -1105,18 +1105,19 @@ private:
   }
 
   void set_currentpoint (int px, int py)
   {
     if (!fp.is_beingdeleted ())
       {
         Matrix pos (1,2,0);
         pos(0) = px;
-        // FIXME - only works if (uimenu->is_visible ()) ?
-        pos(1) = h () - status_h - menu_h - py;
+        pos(1) = h () - menu_h - py;
+        if (! uimenu->is_visible ())
+          pos(1) = pos(1) +  menu_h;
         fp.set_currentpoint (pos);
       }
   }
 
   void set_axes_currentpoint (graphics_object ax, int px, int py)
   {
     if (ax.valid_object ())
       {
